src/NpmUtilities.js

```diff
@@ -157,11 +157,10 @@ export default class NpmUtilities {
       cwd: directory,
     };
 
-    if (registry) {
-      opts.env = Object.assign({}, process.env, {
-        npm_config_registry: registry,
-      });
-    }
+    // pass environment variables to sub commands
```

Comment:
How is this different from what [execa does by default](https://github.com/sindresorhus/execa/blob/dc7e21b0754985b75a5e89917c9035beac1f1f2e/index.js#L19-L26) in v0.8.0? I'd rather not incur the expensive iterating of `process.env` twice per call, if we can help it. We could even get rid of it in the existing conditional, given what execa already does.

A recent investigation of Appveyor failures due to undefined env vars led to these PRs (the root cause mysteriously cleaned up with an updated version of a console emulator):
https://github.com/IndigoUnited/node-cross-spawn/issues/80
https://github.com/kentcdodds/kcd-scripts/pull/5